src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,683 @@
+        String[] configDropinsFileList = configDropinsFolder.list();
+        if (configDropinsFileList == null) {
+            return result;
+        }
+        List<String> configDropinsXmls = new ArrayList<String>();
+        for (String xml : configDropinsFileList){
+            if (xml.endsWith(".xml")) {
+                configDropinsXmls.add(configDropins + "/" +xml);
+            }
+        }
+        if (configDropinsXmls.isEmpty()) {
+            return result;
+        }
+        for (String filename : configDropinsXmls) {
+            result.addAll(getServerXmlFeatures(new File(filename), null));
+        }
+        return result;
+    }
+
+    private static Set<String> getServerXmlFeatures(File serverFile, List<File> parsedXmls) {
+        Set<String> result = new HashSet<String>();
+        List<File> updatedParsedXmls = new ArrayList<File>();
+        updatedParsedXmls.add(serverFile);
+        if (parsedXmls != null){
+            updatedParsedXmls.addAll(parsedXmls);
+        }
+        if (serverFile.exists()) {
+            try {
+                Document doc = new XmlDocument() {
+                    public Document getDocument(File file) throws IOException, ParserConfigurationException, SAXException {
+                        createDocument(file);
+                        return doc;
+                    }
+                }.getDocument(serverFile);
+                Element root = doc.getDocumentElement();
+                NodeList nodes = root.getChildNodes();
+                for (int i = 0; i < nodes.getLength(); i++) {
+                    if (nodes.item(i) instanceof Element) {
+                        Element child = (Element) nodes.item(i);
+                        if ("featureManager".equals(child.getNodeName())) {
+                            result.addAll(parseFeatureManagerNode(child));
+                        } else if ("include".equals(child.getNodeName())){
+                            result = parseIncludeNode(result, serverFile, child, updatedParsedXmls);
+                        }
+                    }
+                }
+            } catch (IOException | ParserConfigurationException | SAXException e) {
+                // just skip this server.xml if it cannot be parsed
+            }
+        }
+        return result;
+    }
+    
+    private static Set<String> parseIncludeNode(Set<String> origResult, File serverFile, Element node,
+            List<File> updatedParsedXmls) {
+        Set<String> result = origResult;
+        String includeFileName = node.getAttribute("location");
+        File includeFile = new File(includeFileName);
+        if (!includeFile.isAbsolute()){
+            includeFile = new File(serverFile.getParentFile().getAbsolutePath(), includeFileName);
+        }
+        if (!updatedParsedXmls.contains(includeFile)){
+            String onConflict = node.getAttribute("onConflict");
+            if (!"ignore".equalsIgnoreCase(onConflict)){
```

Comment:
I mentioned this in a previous comment, but I don't think you addressed it so I'll ask here again: don't you have to first check that there is a conflict? For example, if you set `onConflict="ignore"` doesn't that mean to **ignore if there is a conflict** rather than **always ignore**?

Reply:
For ignore specifically, the presence of a featureManager section (whether it has features or not) in the parent counts as a conflict, so the child will be ignored.  The absence of a featureManager section counts as no conflict, so the child contents are used.  Verified this against installUtility behaviour and added unit tests for this.  Updated the logic so that a null origResult indicates absence of a featureManager section.